var/spack/repos/builtin/packages/micromamba/package.py

```diff
@@ -0,0 +1,153 @@
+        depends_on("cli11@2.2:", type="link")
+
+        # Currently fails with:
+        #
+        #     libarchive/archive_digest.c:191: undefined reference to `mbedtls_sha512_free'
+        #
+        # These shouldn't be necessary,
+        # since they are already in 'libarchive crypto=mbedtls xar=libxml2'
+        # but even adding them doesn't fix it.
+        #
+        #     depends_on("libxml2", type="link")
+        #     depends_on("mbedtls", type="link")
+
+    with when("linkage=dynamic"):
+        # See https://github.com/mamba-org/mamba/blob/micromamba-1.0.0/libmamba/CMakeLists.txt#L423
+        depends_on("libsolv+conda", type="link")
+        depends_on("curl libs=shared", type="link")
+        depends_on("libarchive crypto=mbedtls xar=libxml2", type="link")
+        depends_on("openssl", type="link")
+        depends_on("yaml-cpp", type="link")
+        depends_on("libreproc+cxx+shared", type="link")
+        depends_on("tl-expected@b74fecd", type="link")
+        depends_on("fmt", type="link")
+        depends_on("spdlog", type="link")
+        # See linkage=shared for usage location
+        depends_on("nlohmann-json", type="link")
+        depends_on("cpp-termcolor", type="link")
+        depends_on("cli11@2.2:", type="link")
+
+    with when("linkage=static"):
+        # When linkage is static, BUILD_STATIC=ON
+        # and then
+        # https://github.com/mamba-org/mamba/blob/micromamba-1.0.0/libmamba/CMakeLists.txt#L523
+        # calls libmamba_create_target(libmamba-static STATIC SHARED libmamba)
+        # where the third argument, SHARED, is the deps_linkage
+        # as defined here,
+        # https://github.com/mamba-org/mamba/blob/micromamba-1.0.0/libmamba/CMakeLists.txt#L256
+        # which would use dynamic linkage here,
+        # https://github.com/mamba-org/mamba/blob/micromamba-1.0.0/libmamba/CMakeLists.txt#L420
+        # See linkage=dynamic for what that entails.
+        depends_on("libsolv+conda", type="link")
+        depends_on("curl libs=shared", type="link")
+        depends_on("libarchive crypto=mbedtls xar=libxml2", type="link")
+        depends_on("openssl", type="link")
+        depends_on("yaml-cpp", type="link")
+        depends_on("libreproc+cxx", type="link")
+        depends_on("tl-expected@b74fecd", type="link")
+        depends_on("fmt", type="link")
+        depends_on("spdlog", type="link")
+        depends_on("nlohmann-json", type="link")
+        depends_on("cpp-termcolor", type="link")
+        depends_on("cli11@2.2:", type="link")
+
+    patch("fix-threads.patch")
+
+    def cmake_args(self):
+        # See https://mamba.readthedocs.io/en/latest/developer_zone/build_locally.html#build-micromamba
+        if "linkage=dynamic" in self.spec:
+            linkage = "dynamic"
+        elif "linkage=static" in self.spec:
+            linkage = "static"
+        elif "linkage=full_static" in self.spec:
+            warnings.warn("Full static does not work yet!")
```

Comment:
`tty.warn` is better than `warnings` since that's how all Spack error/warning/debug/info messages are printed. But I wouldn't do this here, I would just add a conflict or even remove the option until we get it working.